gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -483,6 +507,17 @@ private void cleanupStagingData(JobState jobState) {
   }
 
   /**
+   * Combines the specified {@link JobListener} with the {@link #mandatoryJobListeners} for this job. Uses
+   * {@link JobListeners#parallelJobListener(List, int)} to create a {@link CloseableJobListener} that will execute all
+   * the {@link JobListener}s in parallel.
+   */
+  private CloseableJobListener getParallelCombinedJobListener(JobListener jobListener, int numThreads) {
+    List<JobListener> jobListeners = Lists.newArrayList(this.mandatoryJobListeners);
```

Comment:
I tried that but an exception gets thrown in `JobListeners.parallelJobListener` when the method tries to remove `null`s from the given `List` (line 41). I stuck with this approach because it allows the `JobListeners.parallelJobListener` to be defensive against `null` entries.